vault/deadlock.go

```diff
@@ -0,0 +1,21 @@
+// +build deadlock
+
+package vault
+
+import (
+	"github.com/sasha-s/go-deadlock"
+)
+
+// DeadlockMutexW, when the build tag `deadlock` is present, behaves like a
```

Comment:
Small nit, why not call it `DeadlockMutex` (and `DeadlockRWMutex`)?

Reply:
I don't remember what my rationale was, so I'll change it to be the more natural way as you suggest.